shared/src/k8s/job.rs

```diff
@@ -0,0 +1,571 @@
+///     println!("found job: {}", job.metadata.name.unwrap())
+/// }
+/// # }
+/// ```
+///
+/// ```no_run
+/// use akri_shared::k8s::job;
+/// use kube::client::Client;
+/// use kube::config;
+///
+/// # #[tokio::main]
+/// # async fn main() {
+/// let field_selector = Some("spec.nodeName=node-a".to_string());
+/// let api_client = Client::try_default().await.unwrap();
+/// for job in job::find_jobs_with_selector(None, field_selector, api_client).await.unwrap() {
+///     println!("found job: {}", job.metadata.name.unwrap())
+/// }
+/// # }
+/// ```
+pub async fn find_jobs_with_selector(
+    label_selector: Option<String>,
+    field_selector: Option<String>,
+    kube_client: Client,
+) -> Result<ObjectList<Job>, anyhow::Error> {
+    trace!(
+        "find_jobs_with_selector with label_selector={:?} field_selector={:?}",
+        &label_selector,
+        &field_selector
+    );
+    let jobs: Api<Job> = Api::all(kube_client);
+    let job_list_params = ListParams {
+        label_selector,
+        field_selector,
+        ..Default::default()
+    };
+    let result = jobs.list(&job_list_params).await;
+    trace!("find_jobs_with_selector return");
+    Ok(result?)
+}
+
+/// Create Kubernetes Job with given Instance and OwnershipInfo
+///
+/// Example:
+///
+/// ```no_run
+/// use akri_shared::k8s::{
+///     OwnershipInfo,
+///     OwnershipType,
+///     job
+/// };
+/// use akri_shared::akri::instance::{Instance, InstanceSpec};
+/// use kube::client::Client;
+/// use kube::config;
+/// use k8s_openapi::api::batch::v1::JobSpec;
+///
+/// # #[tokio::main]
+/// # async fn main() {
+/// let api_client = Client::try_default().await.unwrap();
+/// let instance_spec = InstanceSpec {
+///     configuration_name: "configuration_name".to_string(),
+///     shared: true,
+///     nodes: Vec::new(),
+///     device_usage: std::collections::HashMap::new(),
+///     broker_properties: std::collections::HashMap::new()
+/// };    
+/// let instance = Instance::new("instance_name", instance_spec);
+/// let job = job::create_new_job_from_spec(
+///     &instance,
+///     OwnershipInfo::new(
+///         OwnershipType::Instance,
+///         "instance_name".to_string(),
+///         "instance_uid".to_string()
+///     ),
+///     "akri.sh/configuration_name",
+///     &JobSpec::default(),"app_name").unwrap();
+/// # }
+/// ```
+pub fn create_new_job_from_spec(
+    instance: &Instance,
+    ownership: OwnershipInfo,
+    resource_limit_name: &str,
+    job_spec: &JobSpec,
+    app_name: &str,
+) -> anyhow::Result<Job> {
+    trace!("create_new_job_from_spec enter");
+    // TODO: Consider optionally enabling podAntiAffinity in this function
+    // (using an instance name label) to ensure only one Job runs on each Node per instance.
+    let instance_name = instance.metadata.name.as_ref().unwrap();
+    let mut labels: BTreeMap<String, String> = BTreeMap::new();
```

Comment:
Curious, why are we using a BTree and not a HashMap?

Reply:
Because that that is the [type for `labels` defined by k8s_openapi](https://docs.rs/k8s-openapi/latest/k8s_openapi/apimachinery/pkg/apis/meta/v1/struct.ObjectMeta.html#structfield.labels) (Rust crate for Kubernetes types)